### PR TITLE
Update parsing of Gitlab [tag] push events

### PR DESCRIPTION
**Pull request by jpopelka**
and some refactoring, see the commit messages

### CODE REVIEW

**Comment by jpopelka** [comment]
recheck

File: packit_service/worker/tasks.py [hunk 1]
```diff
@@ -132,15 +132,11 @@ class BodhiHandlerTaskWithRetry(HandlerTaskWithRetry):
 @celery_app.task(
     name=getenv("CELERY_MAIN_TASK_NAME") or CELERY_DEFAULT_MAIN_TASK_NAME, bind=True
 )
-def process_message(
-    self, event: dict, topic: str = None, source: str = None
-) -> List[TaskResults]:
+def process_message(self, event: dict) -> List[TaskResults]:
     """
```
Comment (mfocko) [inline packit_service/worker/tasks.py:136]: Could you please also switch it to the google style docstrings while you're making changes here?

File: packit_service/worker/parser.py [hunk 2]
```diff
@@ -366,47 +366,34 @@ def get_gitlab_push_common_data(event) -> tuple:
...
 
         Returns:
-            a tuple like (actor, project_url, parsed_url, ref, head_commit, head_commit_sha)
+            a tuple like (actor, project_url, parsed_url, ref, head_commit)
         Raises:
             PackitParserException
         """
-        raw_ref = event.get("ref")
+        if not (raw_ref := event.get("ref")):
+            raise PackitParserException("No ref info from event.")
```
Comment (mfocko) [inline packit_service/worker/parser.py:374]: Is this change intentional? Before it was just a warning.

**Review by mfocko — APPROVED** [review]
lgtm

File: packit_service/worker/parser.py [hunk 3]
```diff
@@ -366,47 +366,34 @@ def get_gitlab_push_common_data(event) -> tuple:
...
-
-        ref = raw_ref[-1]
-        head_commit = commits[-1]
-        head_commit_sha = head_commit["id"]
-
-        if not raw_ref:
-            logger.warning("No commit_id info from event.")
+        # The first item in the list should be the head (newest) commit,
+        # but rather not assume anything and select the "checkout_sha" one.
+        head_commit = next(c for c in commits if c["id"] == checkout_sha)
```
Comment (lachmanfrantisek) [inline packit_service/worker/parser.py:388]: :+1:  for the expliciteness

File: packit_service/worker/tasks.py [hunk 4]
```diff
@@ -132,15 +132,11 @@ class BodhiHandlerTaskWithRetry(HandlerTaskWithRetry):
 @celery_app.task(
     name=getenv("CELERY_MAIN_TASK_NAME") or CELERY_DEFAULT_MAIN_TASK_NAME, bind=True
 )
-def process_message(
-    self, event: dict, topic: str = None, source: str = None
-) -> List[TaskResults]:
+def process_message(self, event: dict) -> List[TaskResults]:
```
Comment (lachmanfrantisek) [inline packit_service/worker/tasks.py:135]: Well spotted! Thanks!

**Review by lachmanfrantisek — APPROVED** [review]

File: packit_service/worker/parser.py (hunk 2 above)
Comment (jpopelka) [reply packit_service/worker/parser.py:374]: Yup, later we `split()` it so if this returned `None` we'd end up with an `AttributeError` later anyway, so rather stop parsing right now.

Thanks for the reviews.
